### PR TITLE
OBG-67 Add security-bypass profile

### PR DESCRIPTION
**Pull request by tsehelnyk**
…zation-Session-Key) validation. Includes "no-signature profile".

### CODE REVIEW

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public AisErrorDecoder aisErrorDecoder() {
     }
 
     @Bean
-    @Profile("!no-signature-filter")
+    @Profile("!no-signature-filter | !security-bypass")
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java:49]: Extract `!no-signature-filter | !security-bypass` to global constant

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public RequestInterceptor requestInterceptorWithSigning() {
     }
 
     @Bean
-    @Profile("no-signature-filter")
+    @Profile("no-signature-filter | security-bypass")
```
Comment (valb3r) [inline fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java:59]: To global constant

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java [hunk 3]
```diff
@@ -27,16 +28,33 @@
...
     @Scope(scopeName = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
     public AuthorizationSessionKeyFromHttpRequest getAuthorizationSessionKeyFromHttpRequest(HttpServletRequest httpServletRequest) {
         log.debug("Incoming request {}", httpServletRequest.getRequestURI());
+        String authCookieValue = getAuthCookieValue(httpServletRequest);
+
+        return new AuthorizationSessionKeyFromHttpRequest(authService.validateTokenAndGetSubject(authCookieValue));
+    }
+
+    @Bean
+    @Profile("security-bypass")
```
Comment (valb3r) [inline opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java:41]: To global constant

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java [hunk 4]
```diff
@@ -27,16 +28,33 @@
     private final TokenBasedAuthService authService;
 
     @Bean
+    @Profile("!security-bypass")
```
Comment (valb3r) [inline opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java:31]: To global constant

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/context/ServiceContextProviderConfig.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import de.adorsys.opba.protocol.facade.services.scoped.RequestScopedProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+public class ServiceContextProviderConfig {
+
+    @Bean(NoRedirectCodeValidationServiceContextProvider.NO_REDIRECT_CODE_VALIDATION)
+    NoRedirectCodeValidationServiceContextProvider serviceContextProviderNoRedirectCodeValidation(
```
Comment (valb3r) [inline opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/context/ServiceContextProviderConfig.java:18]: This bean should be activated only on certain profile

**Review by valb3r — CHANGES_REQUESTED** [review]
Group all beans depending on the configuration (security / no security) into single file.
I.e.
NoSecurityConfig.java
```
@Configuration
@Profile("no-security")
public class NoSecurityConfig {
... No security beans
}
```

SecurityConfig.java
```
@Configuration
@Profile("!no-security")
public class SecurityConfig {
... Security beans
}
```

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java (hunk 1 above)
Comment (tsehelnyk) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java]: moved

File: fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java (hunk 2 above)
Comment (tsehelnyk) [reply fintech-examples/fintech-impl/src/main/java/de/adorsys/opba/fintech/impl/config/FeignConfig.java]: moved

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java (hunk 3 above)
Comment (tsehelnyk) [reply opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java]: moved

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java (hunk 4 above)
Comment (tsehelnyk) [reply opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/config/AuthorizationSessionKeyConfig.java]: moved

File: opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/TokenBasedAuthService.java [hunk 6]
```diff
@@ -38,6 +38,24 @@ public String generateToken(String subject, Duration duration) {
...
+
+    @SneakyThrows
+    public String getSubject(String token) {
+        var jwt = getAndVerifySignedJwtFromString(token);
+
+        return jwt.getJWTClaimsSet().getSubject();
+    }
+
+    @SneakyThrows
+    private SignedJWT getAndVerifySignedJwtFromString(String token) {
```
[the remark this answers is not in the document]
Comment (tsehelnyk) [reply opba-api-security/src/main/java/de/adorsys/opba/api/security/internal/service/TokenBasedAuthService.java]: renamed

File: opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/context/ServiceContextProviderConfig.java (hunk 5 above)
Comment (tsehelnyk) [reply opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/context/ServiceContextProviderConfig.java]: This bean is used by DenyAuthorizationService with any profile

**Review by valb3r — APPROVED** [review]

**Pull request merged**